Ginger/Ginger/UserControlsLib/TextEditor/ValueExpression/ValueExpressionMockDataEditorPage.xaml.cs

```diff
@@ -0,0 +1,325 @@
+limitations under the License. 
+*/
+#endregion
+
+using Amdocs.Ginger.Common;
+using Bogus;
+using Ginger.UserControlsLib.TextEditor.Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Windows.Controls;
+
+namespace Ginger.UserControlsLib.TextEditor.ValueExpression
+{
+    /// <summary>
+    /// Interaction logic for ValueExpressionMockDataEditorPage.xaml
+    /// </summary>
+    public partial class ValueExpressionMockDataEditorPage : Page
+    {
+        SelectedContentArgs mSelectedContentArgs;
+        Context mContext;
+        string mObj;
+        public ValueExpressionMockDataEditorPage(Context context, SelectedContentArgs SelectedContentArgs, string obj, string function, string Locale, string MockExpression)
+        {
+            InitializeComponent();
+            mContext = context;
+            mSelectedContentArgs = SelectedContentArgs;
+            mObj = obj;
+            int datasetStartindex = MockExpression.IndexOf('=');
+            int datasetendindex = MockExpression.IndexOf('(');
+            int localestartindex = MockExpression.IndexOf('"');
+            int localeendindex = MockExpression.IndexOf(')');
+            int functionstartindex = MockExpression.IndexOf('.');
+
+            List<string> Localelst = GetLocales();
+            Type objType;
+            List<string> objClasses = new List<string>();
+            List<string> methodList = new List<string>();
+
+            Assembly assembly = Assembly.Load("Bogus"); // or load your target assembly
+
+            string namespaceName = "Bogus.DataSets";
+
+            objClasses = GetObjectClasses(assembly, namespaceName);
+            
+
+            // Example of fetching locales (replace with actual implementation)
+            Localelst = GetLocales();
+
+            string objTypeName = mObj.Equals("Randomizer") ? $"Bogus.{mObj}" : $"Bogus.DataSets.{mObj}";
+            objType = assembly.GetType(objTypeName);
+
+            methodList = GetMethodsOfType(objType);
+            int CaretPossition = SelectedContentArgs.GetCaretPosition();
+            if ((datasetStartindex < CaretPossition) && (CaretPossition < datasetendindex))
+            {
+                FunctionsList.ItemsSource = objClasses.OrderBy(x => x).ToList();
+            }
+            else if ((localestartindex < CaretPossition) && (CaretPossition < localeendindex) && (!mObj.Equals("Randomizer") || !mObj.Equals("Finance")))
+            {
+                FunctionsList.ItemsSource = Localelst.OrderBy(x => x).ToList();
+            }
+            else
+            {
+                FunctionsList.ItemsSource = methodList.OrderBy(x => x).ToList();
+            }
+        }
+        
+            
+        private static List<string> GetLocales()
+        {
+            // Replace with actual implementation to fetch locales
+            return Bogus.Database.GetAllLocales().ToList();
```

Comment:
try to maintain static list of locale and reuse it if possible

Reply:
Done